### PR TITLE
added component has permissions

### PR DESCRIPTION
**Pull request by douglascamposh**

### CODE REVIEW

File: src/components/permissions/HasPermission.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      const claims = JSON.parse(atob(token.split('.')[1]));
+      const pagePermissions = rolesData.filter(({roleName}) => claims.roles.find(currentRole => currentRole === roleName)).flatMap(({permissions}) => permissions);
+      const hasAccess = pagePermissions.find(({page}) => pathName.includes(page));
+      return hasAccess && hasAccess.methods.find(method => permissionTo === method) ? (<>{children}</>) : (<></>);
+    }
+    
+    if (isLoading) {
+      return (
+        <div>
+          <Spinner></Spinner>
```
Comment (mgamarrav) [inline src/components/permissions/HasPermission.js]: change this line only like `<Spinner/>`

File: src/components/permissions/HasPermission.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        </div>
+      );
+    }
+    if (isError) {
+      return (
+        <div>
+          <ServerErrorComponent></ServerErrorComponent>
```
Comment (mgamarrav) [inline src/components/permissions/HasPermission.js]: the same here  eg: `<ServerErrorComponent />`

File: src/components/permissions/HasPermission.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import { usePathname } from "next/navigation";
+import { methods } from "@/app/constants/constants";
+
+const HasPermission = ({children, to:permissionTo=methods.read}) => {
+    const { data: rolesData, isLoading, isError } = useGetRolesQuery();
+    const pathName = usePathname();
+    const token = localStorage.getItem('token');
+    if(token && rolesData) {
+      const claims = JSON.parse(atob(token.split('.')[1]));
+      const pagePermissions = rolesData.filter(({roleName}) => claims.roles.find(currentRole => currentRole === roleName)).flatMap(({permissions}) => permissions);
```
Comment (mgamarrav) [inline src/components/permissions/HasPermission.js]: this line is too long maybe you could consider to separate in 2 lines

File: src/components/permissions/HasPermission.js (hunk 1 above)
Comment (douglascamposh) [reply src/components/permissions/HasPermission.js]: ok

File: src/components/permissions/HasPermission.js (hunk 2 above)
Comment (douglascamposh) [reply src/components/permissions/HasPermission.js]: ok

File: src/components/permissions/HasPermission.js (hunk 3 above)
Comment (douglascamposh) [reply src/components/permissions/HasPermission.js]: ok

**Review by mgamarrav — APPROVED** [review]

**Pull request merged**